code/auth-service/src/dataaccess/userRolesStorageRepository.js

```diff
@@ -0,0 +1,19 @@
+import database from '../config/database';
+
+function UserRoles() {
+  return database.getModel('UserRoles');
+}
+
+function getRoles(userId) {
+  return new Promise((resolve, reject) => {
+    const query = UserRoles().findOne({ userId });
+    query.exec((err, document) => {
+      if (err) {
```

Comment:
I don't think you need to handle the error here as `exec()` returns a promise. I think it can just be

```
function getRoles(userId) {
  return UserRoles().findOne({ userId }).exec();
}
```